### DISCUSSION TITLE
How to speedup bitbucket get pr list

### DISCUSSION THREAD

**Discussion by monwolf** [renovatebot/renovate#26266] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.91

### If you're self-hosting Renovate, select which platform you are using.

Bitbucket Cloud (bitbucket.org)

### What is your question?

I've been using Renovate for a while and I'm planning to extend it to other teams, I realized Renovate every time spends more time for each project. 
Tailing the logs, I could observe most of the time is wasted in fetching PRs from bitbucket API, this project is small in size but It has about 1500 PRs (closed/merged) from renovatebot. 

```
DEBUG: findPr(renovate/configure, Configure Renovate, !open) (repository=org/my-front)
       "dd": {"service": "renovate"}
DEBUG: getPrList() (repository=org/my-front)
       "dd": {"service": "renovate"}
DEBUG: Retrieving PR list (repository=org/my-front)
       "dd": {"service": "renovate"}

```
More than 3min is "stuck" in this step. Is there any way to speed up this? 

### Logs (if relevant)

<details><summary>Logs</summary>

```

Copy/paste the relevant log(s) here, between the starting and ending backticks

```

</details>

**Comment by rarkins** [comment]
We probably could/should implement like we did for GitHub - cache the list of PRs and then on subsequent runs only fetch as many pages as we need to get the updated ones, which is usually only one page.

Can you check if the BB API returns or can return PRs based on "last modified date"?

**Reply by monwolf** [reply]
We can filter it, This should be the syntax

```
?q=updated_on > 2022-11-11T00:00:00-07:00
```
Example (with results):
![image](https://github.com/renovatebot/renovate/assets/1116133/36c40446-24c6-4e0d-9d78-926ed44efb83)

Example (without results):
![image](https://github.com/renovatebot/renovate/assets/1116133/51f85eb5-a78f-49d9-a4be-bac242d6bb68)


I checked the API and if we know which fields Renovatebot needs to work on, we could remove non-used noisy fields:

For Example: 
```
&fields=-values.description,-values.links,-values.summary
```

**Reply by rarkins** [reply]
That sounds excellent. Can you create a new "Suggest an idea" discussion to supersede this one which says to cache and reuse PR lists on Bitbucket Cloud and includes your suggestion on using the updated_on and fields options? I will then convert it into a feature request issue

**Reply by monwolf** [reply, marked as answer]
Here it is https://github.com/renovatebot/renovate/discussions/26275
